### PR TITLE
Jar packaging and build cleanup

### PR DESCRIPTION
**Pull request by caalador**
Jar packaging.

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/skeleton-starter-flow-spring/77)

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -15,13 +16,20 @@
         <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
         <failOnMissingWebXml>false</failOnMissingWebXml>
 
-        <webapp.directory>src/main/webapp</webapp.directory>
+        <transpilation.directory>${project.build.directory}/classes/META-INF/resources</transpilation.directory>
```
Comment (mstahv) [inline pom.xml]: Why add these as properties if they are only used in one place?

File: pom.xml [hunk 2]
```diff
@@ -150,12 +130,12 @@
                                     <goal>package-for-production</goal>
                                 </goals>
                                 <configuration>
-                                    <transpileOutputDirectory>${webapp.directory}</transpileOutputDirectory>
+                                    <transpileOutputDirectory>${transpilation.directory}</transpileOutputDirectory>
```
Comment (mstahv) [inline pom.xml]: Ideally maven plugin could detect that this is a spring boot project and have these as defaults.

File: pom.xml [hunk 3]
```diff
@@ -65,11 +66,6 @@
             <artifactId>vaadin-spring-boot-starter</artifactId>
             <version>${vaadin.version}</version>
         </dependency>
-
```
Comment (mstahv) [inline pom.xml]: To remove this, you need an updated vaadin-spring-boot-starter. The rc1 don't have vaadin-spring dependency.

Comment (pleku) [reply pom.xml]: We will probably make an RC2 tomorrow with all the changes in flow & maven-plugin & vaadin-spring

Comment (mstahv) [reply pom.xml]: master branches could point to snapshot versions, so we could test theme before new release?

**Comment by caalador** [comment]
That's not possible as vaadin.com/start uses the tagged revision from github so it needs to be "correct"

Also what you need right now to test it is:
```

        <dependency>
            <groupId>com.vaadin</groupId>
            <artifactId>vaadin-spring-boot-starter</artifactId>
            <version>10.0-SNAPSHOT</version>
        </dependency>

        <dependency>
            <groupId>com.vaadin</groupId>
            <artifactId>vaadin-spring</artifactId>
            <version>10.0-SNAPSHOT</version>
        </dependency>

        <dependency>
            <groupId>com.vaadin</groupId>
            <artifactId>flow-server</artifactId>
            <version>1.0-SNAPSHOT</version>
        </dependency>

        <dependency>
            <groupId>com.vaadin</groupId>
            <artifactId>flow-client</artifactId>
            <version>1.0-SNAPSHOT</version>
        </dependency>
```

---

Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks broke.

---



*Comments from [Reviewable](https://reviewable.io/reviews/vaadin/skeleton-starter-flow-spring/77#-:-LEEpzzCD7w8IX0GjAJe:bpkdaop)*

**Comment by mstahv** [comment]
@caalador, naming (and documentation) should be clear in the configuration section as well. To me it becomes quite weird if I need to jump from the plugin configuration back to the properties section to see what the actual value is. For the property I get absolutely no documentation at all, unlike for the configuration section (note that our documentation don't currently work, but it should. thus the example screenshot below is from spring-boot plugins docs):

screen shot 2018-06-05 at 15 06 24

I haven't thought about how vaadin-maven-plugin could detect spring-boot project, but I guess the maven plugin could check if spring boot dependencies exist. Any ideas @snicoll?

**Comment by mstahv** [comment]
vaadin.com/start could choose a certain tag or branch, rather than master?

**Comment by mstahv** [comment]
@caalador Here is one idea how vaadin/flow-maven-plugin could work without extra effort in spring-boot projects: https://github.com/vaadin/flow/compare/master...mstahv:features/detect-boot?expand=1

**Comment by caalador** [comment]
Dismissed @mstahv from 2 discussions.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks broke.

---



*Comments from [Reviewable](https://reviewable.io/reviews/vaadin/skeleton-starter-flow-spring/77)*

**Comment by mstahv** [comment]
Some additional cleanup for this branch in #79 

The title of this PR should be "Jar packaging and build cleanup" or similar.

**Comment by ZheSun88** [comment]
Reviewed 4 of 4 files at r4.
Review status: :shipit: complete! all files reviewed, all discussions resolved

---



*Comments from [Reviewable](https://reviewable.io/reviews/vaadin/skeleton-starter-flow-spring/77)*

**Pull request merged**